src/tuwien/auto/calimero/knxnetip/util/Srp.java

```diff
@@ -0,0 +1,288 @@
+			throw new KNXFormatException("buffer too short for SRP header");
+
+		size = data[offset] & 0xff;
+		if (size > data.length - offset)
+			throw new KNXFormatException("SRP size bigger than actual data length", size);
+
+		mandatory = (data[offset + 1] & 0x80) == 0x80;
+		type = Srp.Type.from(data[offset + 1] & 0x7f);
+
+		if ((size - SrpHeaderSize) > 0) {
+			this.data = new byte[size - SrpHeaderSize];
+			System.arraycopy(data, offset + SrpHeaderSize, this.data, 0, size - SrpHeaderSize);
+		} else {
+			this.data = new byte[0];
+		}
+	}
+
+	/**
+	 * Returns a search request parameter block to limit the extended search request to KNXnet/IP router or
+	 * server devices where programming mode is currently enabled. The mandatory flag of the SRP is not set.
+	 *
+	 * @return search request parameter block for devices currently in programming mode
+	 */
+	public static Srp inProgrammingMode() {
+		return new Srp(Type.SelectByProgrammingMode, false);
+	}
+
+	/**
+	 * Returns a search request parameter block to limit the extended search request to KNXnet/IP router
+	 * or server devices where programming mode is currently enabled. The mandatory flag of the SRP is set.
+	 *
+	 * @return mandatory search request parameter block for devices currently in programming mode
+	 */
+	public static Srp inProgrammingModeOnly() {
+		return new Srp(Type.SelectByProgrammingMode, true);
+	}
+
+	public static Srp withMacAddress(final byte[] macAddress) {
+		return new Srp(Type.SelectByMacAddress, false);
+	}
+
+	public static Srp withMacAddressOnly(final byte[] macAddress) {
+		return new Srp(Type.SelectByMacAddress, true);
+	}
+
+	public static Srp withService(final int familyId, final int familyVersion) {
+		return new Srp(Type.SelectByService, false, (byte) familyId, (byte) familyVersion);
+	}
+
+	public static Srp withServiceOnly(final int familyId, final int familyVersion) {
+		return new Srp(Type.SelectByService, true, (byte) familyId, (byte) familyVersion);
+	}
+
+	public static Srp withDeviceDescription(final int descriptionType, final int... additionalDescriptionTypes) {
+		final ByteBuffer buffer = ByteBuffer.allocate(additionalDescriptionTypes.length + 1);
+		for (final int dt : additionalDescriptionTypes)
+			buffer.put((byte) dt);
+		buffer.put((byte) descriptionType);
+
+		return new Srp(Type.RequestDibs, false, buffer.array());
+	}
+
+	/**
+	 * Returns the structure length of this SRP in bytes.
+	 * <p>
+	 *
+	 * @return structure length as unsigned byte
+	 */
+	public final int getStructLength() {
+		return size;
+	}
+
+	/**
+	 * Returns the type of this SRP.
+	 * <p>
+	 * The type specifies which kind of search request parameter information is contained in
+	 * the SRP.
+	 *
+	 * @return search request parameter type {@see Type}
+	 */
+	public final Srp.Type getType() {
+		return type;
+	}
+
+	/**
+	 * Returns the mandatory flag of this SRP.
+	 * <p>
+	 *
+	 * @return <code>true</code> if the mandatory bit is set, <type>false</type> otherwise
```

Comment:
`<type>` does not do anything, either `<code>` or `{@code ...}` need to be used to show formatting

Reply:
Fixed.